decide/decide.cpp

```diff
@@ -36,6 +36,30 @@ PUM calculatePUM(CMV cmv, LCM lcm) {
   return pum;
 }
 
+FUV calculateFUV(PUM &pum, PUV &puv) {
+  int n = pum.size();
+  FUV fuv(n);
+
+  for (int i = 0; i < n; i++) {
+    fuv[i] = true;
+
+    // FUV[i] should be set to true if PUV[i] is false ...
+    if (!puv[i]) {
+      fuv[i] = true;
+      continue;
+    }
+
+    // ... or if all elements in row PUM[i] are true.
+    for (int j = 0; j < n; j++) {
+      if (pum[i][j]) {
```

Comment:
This should be `if (!pum[i][j])`, right?

Reply:
Very good catch!